execution/src/windows/execution.cc

```diff
@@ -221,17 +222,53 @@ namespace leatherman { namespace execution {
         const string name;
         HANDLE handle;
         OVERLAPPED overlapped;
-        scoped_resource<HANDLE> event;
+        scoped_handle event;
         bool pending;
         string buffer;
         function<bool(string const&)> const& callback;
     };
 
-    static void read_from_child(DWORD child, array<pipe, 2>& pipes, uint32_t timeout, HANDLE timer)
+    struct input_pipe
+    {
+        scoped_handle handle;
+        string buffer;
+        bool pending;
+    };
+
+    static void rw_from_child(DWORD child, array<pipe, 2>& pipes, input_pipe input, uint32_t timeout, HANDLE timer)
```

Comment:
@joshcooper @peterhuene I could mainly use help confirming the changes to this function make sense. They seem to work, but I might be missing some corner cases.

I'm currently not using OVERLAPPED for the write, and still using a file handle. I think both of those things are ok because the child process isn't going to be attempting to read and write at the same time (probably?)